include/jobs.hpp

```diff
@@ -62,22 +62,21 @@ namespace NP {
 			key = (key << 4) ^ h(arrival.from());
 			key = (key << 4) ^ h(id.task);
 			key = (key << 4) ^ h(arrival.until());
-			key = (key << 4) ^ h(exec_time[0].from());
+			key = (key << 4) ^ h(exec_time.begin()->second.from());
 			key = (key << 4) ^ h(deadline);
-			key = (key << 4) ^ h(exec_time[0].upto());
+			key = (key << 4) ^ h(exec_time.begin()->second.upto());
 			key = (key << 4) ^ h(id.job);
 			key = (key << 4) ^ h(priority);
 		}
 
 	public:
 
 		Job(unsigned long id,
-			Interval<Time> arr, Cost exec_time,
-			Interval<unsigned int> parallelism,
+			Interval<Time> arr, const Cost& costs,
 			Time dl, Priority prio,
 			Job_index idx,
 			unsigned long tid = 0)
-		: arrival(arr), exec_time(exec_time), parallelism(parallelism),
+		: arrival(arr), exec_time(costs), parallelism(costs.begin()->first, costs.rbegin()->first),
```

Comment:
Is there a particular reason you're using the reverse iterator `rbegin()` instead of `end()`? I find `end()` to be more readable.

Reply:
> Is there a particular reason you're using the reverse iterator `rbegin()` instead of `end()`? I find `end()` to be more readable.

If I am not mistaken `x.rbegin()` would be equivalent to `--x.end()`. It avoids having to decrement the iterator. No major reason otherwise.